br/pkg/restore/snap_client/pipeline_items.go

```diff
@@ -106,6 +107,125 @@ func defaultOutputTableChan() chan *CreatedTable {
 	return make(chan *CreatedTable, defaultChannelSize)
 }
 
+// Exhaust drains all remaining errors in the channel, into a slice of errors.
+func Exhaust(ec <-chan error) []error {
+	out := make([]error, 0, len(ec))
+	for {
+		select {
+		case err := <-ec:
+			out = append(out, err)
+		default:
+			// errCh will NEVER be closed(ya see, it has multi sender-part),
+			// so we just consume the current backlog of this channel, then return.
+			return out
+		}
+	}
+}
+
+type PipelineContext struct {
+	// pipeline item switch
+	Checksum         bool
+	LoadStats        bool
+	WaitTiflashReady bool
+
+	// pipeline item configuration
+	LogProgress         bool
+	ChecksumConcurrency uint
+	StatsConcurrency    uint
+
+	// pipeline item tool client
+	KvClient   kv.Client
+	ExtStorage storage.ExternalStorage
+	Glue       glue.Glue
+}
+
+// RestorePipeline do some work in pipeline, such as checkum, load stats and wait tiflash ready.
```

Comment:
just a nit, I feel it's better to describe what a method does precisely instead of being ambiguous, for example we could say 
```
RestorePipeline does checksum, load stats and wait for tiflash to be ready.
```
The advantage of that is people don't need to read the code to understand what it does. Just a thought

Reply:
Done